packages/@lwc/engine-core/src/framework/mutation-tracker.ts

```diff
@@ -28,10 +29,29 @@ export function componentValueMutated(vm: VM, key: PropertyKey) {
     }
 }
 
-export function componentValueObserved(vm: VM, key: PropertyKey) {
+export function componentValueObserved(vm: VM, key: PropertyKey, target: any = {}) {
+    const { component, tro } = vm;
     // On the server side, we don't need mutation tracking. Skipping it improves performance.
     if (process.env.IS_BROWSER) {
-        valueObserved(vm.component, key);
+        valueObserved(component, key);
+    }
+
+    // The portion of reactivity that's exposed to signals is to subscribe a callback to re-render the VM (templates).
+    // We check check the following to ensure re-render is subscribed at the correct time.
+    //  1. The template is currently being rendered (there is a template reactive observer)
+    //  2. There was a call to a getter to access the signal (happens during vnode generation)
+    if (
+        lwcRuntimeFlags.ENABLE_EXPERIMENTAL_SIGNALS &&
+        target &&
+        typeof target === 'object' &&
+        'value' in target &&
+        'subscribe' in target &&
+        typeof target.subscribe === 'function' &&
+        // Only subscribe if a template is being rendered by the engine
+        tro.isObserving()
+    ) {
+        // Subscribe the template reactive observer's notify method, which will mark the vm as dirty and schedule hydration.
+        subscribeToSignal(component, target, tro.notify.bind(tro));
```

Comment:
You might be wondering why I'm exporting and using the `signal-tracker` inside the `mutation-tracker` in `@lwc/engine-core` when I was trying to keep the `mutation-tracker` library clean.

The reason is because the only real function I could see exporting from a `signal-tracker` module in `@lwc/engine-core` was something that would export a similar function as `valueObserved`. 

Rather than create a separate function, I decided to reuse it since it's already aware of things outside the `mutation-tracker` library, like the `vm`.

Reply:
```suggestion
        subscribeToSignal(component, target, FunctionBind.call(tro.notify, tro));
```

Reply:
@nolanlawson it looks like we don't have a `FunctionBind` extracted in `@lwc/shared`. 

I plan to open a separate PR to expose some of the common function methods and I'll update it here as part of the PR.